step/step.go

```diff
@@ -1,42 +1,72 @@
 package step
 
 import (
+	"encoding/json"
 	"fmt"
 	"os"
+
+	"github.com/stackpulse/steps-sdk-go/env"
 )
 
 const (
-	ExitCodeOK      = 0
-	ExitCodeFailure = 1
+	stepErrorKey = "step_error"
 )
 
 type Outputs struct {
 	Object interface{} `json:"api_object,omitempty"`
 }
 
+type JSONOutput map[string]interface{}
+
 // An interface describing a general purpose stackpulse step
 type Step interface {
-	Init() error
-	Run() (exitCode int, output []byte, err error)
+	Init() *Error
+	Run() (JSONOutput, *Error)
 }
 
-// A helper method that execute a given step according to step lifecycle
-func Run(step Step) {
-	err := step.Init()
-	if err != nil {
-		fmt.Fprintf(os.Stderr, "failed init step arguments, %v\n", err)
-		os.Exit(ExitCodeFailure)
+func printOutput(stepOutput map[string]interface{}, stepError *Error) {
+	if stepError == nil {
+		stepError = &Error{
+			Code:    OK,
+			Message: "success",
+		}
 	}
 
-	exitCode, output, err := step.Run()
+	if stepOutput == nil {
+		stepOutput = map[string]interface{}{}
+	}
+
+	stepOutput[stepErrorKey] = *stepError
+
+	JSONStepOutput, err := json.Marshal(stepOutput)
 	if err != nil {
-		fmt.Fprintf(os.Stderr, "Step error: %s\n", err.Error())
-		if output != nil {
-			os.Stderr.Write([]byte("\n"))
-			os.Stderr.Write(output)
+		stepError = &Error{
+			Code:    InvalidArgument,
+			Message: "something went wrong. can't parse step output",
+			Verbose: fmt.Errorf("error: %w", err).Error(),
 		}
-		os.Exit(exitCode)
+
+		stepOutput[stepErrorKey] = *stepError
 	}
 
-	os.Stdout.Write(output)
+	os.Stdout.Write([]byte(fmt.Sprintf("%s%s", env.EndMarkerDefault, JSONStepOutput)))
```

Comment:
more efficient in this case
```suggestion
	os.Stdout.Write([]byte(env.EndMarkerDefault + jsonStepOutput))
```